File: 01-JavaScript/10-fibonacci-generator.js
const prompt = require("prompt-sync")({ sigint: true });

let fibonacciNumbers = [];

function fibonacciGenerator() {
    const numItems = prompt("Enter a whole number and get the list of Fibonacci's numbers: ");

    if (numItems === 1) {
        fibonacciNumbers.push(0);

    } else if (numItems === 2) {
        fibonacciNumbers.push(0, 1);

    } else {
        fibonacciNumbers.push(0, 1);

        for (let count = 2; count < numItems; count++) {
            const lastItem = fibonacciNumbers[fibonacciNumbers.length - 1];
            const beforeLastItem = fibonacciNumbers[fibonacciNumbers.length - 2];
            const nextItem = lastItem + beforeLastItem;
            fibonacciNumbers.push(nextItem);
        }
    }

    console.log(fibonacciNumbers);
}

fibonacciGenerator();

File: 02-Python/10-fibonacci-generator.py
#!/usr/bin/python

fibonacci_numbers = []

def fibonacci_generator():
    num_items = int(input("Enter a whole number and get the list of Fibonacci's numbers: "))

    if num_items == 1:
        fibonacci_numbers.append(0)
    
    elif num_items == 2:
        fibonacci_numbers.extend([0, 1])
    
    else:
        fibonacci_numbers.extend([0, 1])

        for _ in range(2, num_items):
            last_item = fibonacci_numbers[len(fibonacci_numbers)-1]
            before_last_item = fibonacci_numbers[len(fibonacci_numbers)-2]
            next_item = last_item + before_last_item
            fibonacci_numbers.append(next_item)

    print(fibonacci_numbers)



fibonacci_generator()